File: 프로그래머스/0/181836. 그림 확대/그림 확대.swift
import Foundation

func solution(_ picture:[String], _ k:Int) -> [String] {
    var result = picture.flatMap {
        Array(repeating: $0, count: k)
    }
    
    // 문자열을 리턴해야함
    return result.map { str in
        // ".xx...xx."
        return str.map { ch in String(Array(repeating: ch, count: k)) }.joined()
    }
}

File: 프로그래머스/0/181836. 그림 확대/그림 확대.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

string makeWidth(string str, const int &k) {
    string result = "";

    for (char ch : str) {
        for (int i = 0; i < k; i++) {
            result += ch;
        }
    }

    return result;
}

vector<string> makeHeight(vector<string> v, const int &k) {
    vector<string> result;

    for (string str : v) {
        for (int i = 0; i < k; i++) {
            result.push_back(str);
        }
    }

    return result;
}

vector<string> solution(vector<string> picture, int k) {
    vector<string> answer;

    for (int i = 0; i < picture.size(); i++) {
        picture[i] = makeWidth(picture[i], k);
    }

    answer = makeHeight(picture, k);

    return answer;
}